query-engine/query-engine-c-abi/src/lib.rs

```diff
@@ -0,0 +1,9 @@
+pub mod engine;
+pub mod error;
+pub mod functions;
+pub mod logger;
+pub mod migrations;
+
+mod tracer;
+
+pub(crate) type Result<T> = std::result::Result<T, error::ApiError>;
```

Comment:
This type is also exported from `query-engine-common`, and `query-engine-c-abi/src/engine.rs` should be using the common type.

Also, most of the `engine.rs` functions should ideally be using the common's `Result` type. Concretely, that means instead of using `Result<T, Error>`, you should use `query_engine_common::Result<T>`.

Reply:
Got it, done